### ISSUE TITLE
Crash at xyz.aprildown.ultimateringtonepicker.UtilsKt.getParcelableExtraCompat(Utils.kt:95)

### ISSUE THREAD

**Issue by cempo** [DeweyReed/UltimateMusicPicker#34]
```
val settings = UltimateRingtonePicker.Settings(
        systemRingtonePicker = UltimateRingtonePicker.SystemRingtonePicker(
            customSection = UltimateRingtonePicker.SystemRingtonePicker.CustomSection(useSafSelect = true),
            defaultSection = UltimateRingtonePicker.SystemRingtonePicker.DefaultSection(
                showSilent = false,
                defaultUri = defaultRingtoneUri
            ),
            ringtoneTypes = listOf(
                RingtoneManager.TYPE_ALARM,
                RingtoneManager.TYPE_RINGTONE,
                RingtoneManager.TYPE_NOTIFICATION
            )
        )
    )

val intent = RingtonePickerActivity.getIntent(
        context = context,
        settings = settings,
        windowTitle = "Title"
    )
```

Crash happens on different android versions and manufacturers

Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Class.isInterface()' on a null object reference
       at java.lang.Class.isAssignableFrom(Class.java:824)
       at android.os.Parcel.readParcelableCreatorInternal(Parcel.java:4865)
       at android.os.Parcel.readParcelableInternal(Parcel.java:4778)
       at android.os.Parcel.readValue(Parcel.java:4544)
       at android.os.Parcel.readValue(Parcel.java:4324)
       at android.os.Parcel.-$$Nest$mreadValue()
       at android.os.Parcel$LazyValue.apply(Parcel.java:4422)
       at android.os.Parcel$LazyValue.apply(Parcel.java:4381)
       at android.os.BaseBundle.getValueAt(BaseBundle.java:394)
       at android.os.BaseBundle.getValue(BaseBundle.java:374)
       at android.os.BaseBundle.getValue(BaseBundle.java:357)
       at android.os.BaseBundle.get(BaseBundle.java:696)
       at android.os.Bundle.getParcelable(Bundle.java:947)
       at android.content.Intent.getParcelableExtra(Intent.java:9558)
       at xyz.aprildown.ultimateringtonepicker.UtilsKt.getParcelableExtraCompat(Utils.kt:95)
       at xyz.aprildown.ultimateringtonepicker.RingtonePickerActivity.onCreate(RingtonePickerActivity.kt:27)
       at android.app.Activity.performCreate(Activity.java:8591)
       at android.app.Activity.performCreate(Activity.java:8570)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1384)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4150)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4325)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2574)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8762)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)

**Comment by DeweyReed** [comment]
Thanks for your feedback! 

Can you reproduce the crash? I have some solutions, but I need to determine if they work.

**Comment by cempo** [comment]
I'm unable to replicate this crash as I only have crash logs from users

**Comment by DeweyReed** [comment]
Do they crash only on Android T(13) and later, or all system versions?

The code is https://github.com/DeweyReed/UltimateRingtonePicker/blob/master/library/src/main/java/xyz/aprildown/ultimateringtonepicker/Utils.kt#L95.

**Comment by cempo** [comment]
I see crashes for Android 8, 11, 13
<img width="1225" alt="Screenshot 2024-01-11 at 08 18 16" src="https://github.com/DeweyReed/UltimateRingtonePicker/assets/3032189/5802954a-4793-4e8f-ac7d-b308d5a72b60">
<img width="1224" alt="Screenshot 2024-01-11 at 08 18 38" src="https://github.com/DeweyReed/UltimateRingtonePicker/assets/3032189/5bf3131f-3a08-4dbb-8383-b5cb2592d013">

**Comment by DeweyReed** [comment]
This is a weird crash. The only solution I can think of is to wrap the code into a try-catch and finish the activity if it goes wrong.
At the moment, you can copy the activity into your project and modify it as needed.

**Comment by DeweyReed** [comment]
Are you using AGP 8.1 or above? It's probably because of proguard rules.

**Comment by cempo** [comment]
Yes. AGP 8.2. What ProGuard rules are required?

**Comment by DeweyReed** [comment]
I released [a new version 3.3.0-RC1](https://github.com/DeweyReed/UltimateRingtonePicker/releases/tag/3.3.0-RC1), and upgraded all dependencies, but I'm unsure if it fixes the problem.

**Comment by cempo** [comment]
Thank you for the update. I can provide some feedback later after testing the app with real users

**Comment by DeweyReed** [comment]
This is probably the cause: [The new Intent.getParcelableExtra(String,Class) throws an NPE internally](https://issuetracker.google.com/issues/240585930#comment6). The latest release has been using the method from Jetpack. I haven't been able to test it with my app due to lack of time.

**Comment by DeweyReed** [comment]
A user reported that the crash was gone. I released a version(3.3.0) that is identical to RC1.
